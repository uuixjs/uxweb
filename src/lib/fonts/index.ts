// export * from "./sc-font-declaration";
