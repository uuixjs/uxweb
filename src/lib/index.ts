// export * from './fonts';
export * from './pure-color';
// export * from './test-utils';
// export * from './ui-scss-compat';
export * from './ui-tokens';
export * from './ui-utils';
export * from './utils-ts';
export * from './utils-styled-components';
