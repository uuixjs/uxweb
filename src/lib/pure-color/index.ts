import convert from "./convert";
import parse from "./parse";

export * from "./types";
export * from "./convert";
export * from "./parse";

export const pureColor = {
  convert,
  parse,
};

export default pureColor;
