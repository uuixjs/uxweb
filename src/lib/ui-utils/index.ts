export * from "./accent-color-generator";
export * from "./aria-props";
export * from "./classnames";
export * from "./color";
export * from "./data-props";
export * from "./unique-id";
export * from "./root-selectors";

// re-export of sub-package
export * from "../utils-styled-components";
