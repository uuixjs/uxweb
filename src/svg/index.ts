export * from "./component";
export * from "./paths";
