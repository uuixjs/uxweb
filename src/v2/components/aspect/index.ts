export { Align, Aspect, AspectProps, AspectRatio } from "./component";
