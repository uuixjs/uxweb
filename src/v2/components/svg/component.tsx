import { SVGAsset, SVG_PATHS } from "./paths";
import {
  getAriaProps,
  getDataProps,
  styled,
  themeTokenRule,
} from "lib/ui-utils";

import { FC } from "react";
import { ThemeToken } from "lib/ui-tokens";

export enum SVGType {
  Inherit = "color-fill-current",
  Alt = "color-fill-alt",
  Alt2 = "color-fill-alt-2",
  Brand = "color-fill-brand",
  Live = "color-text-accessible-red",
  Warn = "color-fill-warn",
  Alert = "color-fill-alert",
  Success = "color-fill-success",
  Prime = "color-fill-prime",
  Placeholder = "color-background-placeholder",
  OverlayPlaceholder = "color-background-overlay-placeholder",
}

interface ScFigureProps {
  size: "100%" | undefined;
}

const ScFigure = styled.figure<ScFigureProps>`
  align-items: center;
  display: inline-flex;
  height: ${({ size }) => size};
  width: ${({ size }) => size};
`;

interface ScSvgProps {
  /**
   * Forces all values in the SVGType enum to be valid ThemeToken names
   */
  type: ThemeToken;
}

const ScSvg = styled.svg<ScSvgProps>`
  fill: ${({ type }) => themeTokenRule(type)};
`;

export interface SVGProps {
  asset: SVGAsset;
  height?: number;
  width?: number;
  /**
   * Deprecated alias for fillParent.
   *
   * @deprecated
   */
  fill?: boolean;
  /**
   * Controls whether the icon expands to fill its parent's size.
   */
  fillParent?: boolean;
  type?: SVGType;
}

export const SVG: FC<SVGProps> = (props) => {
  const fillParent = props.fillParent || props.fill;

  return (
    <ScFigure
      className="tw-svg"
      size={fillParent ? "100%" : undefined}
      {...getDataProps(props)}
      {...getAriaProps(props)}
    >
      <ScSvg
        type={props.type || SVGType.Inherit}
        width={fillParent ? "100%" : (props.width || 20) + "px"}
        height={fillParent ? "100%" : (props.height || 20) + "px"}
        version="1.1"
        viewBox={`0 0 ${SVG_PATHS[props.asset].width} ${
          SVG_PATHS[props.asset].height
        }`}
        x="0px"
        y="0px"
      >
        {SVG_PATHS[props.asset].path}
      </ScSvg>
    </ScFigure>
  );
};

SVG.displayName = "SVG";
