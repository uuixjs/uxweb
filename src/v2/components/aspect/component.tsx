import { FC, ReactNode } from "react";

import { styled } from "lib/ui-utils";

export enum AspectRatio {
  Aspect21x9 = "tw-aspect--21x9",
  Aspect16x9 = "tw-aspect--16x9",
  Aspect4x3 = "tw-aspect--4x3",
  Aspect1x1 = "tw-aspect--1x1",
  Aspect3x4 = "tw-aspect--3x4",
  Aspect3x2 = "tw-aspect--3x2",
  // Leaving BoxArt as a semantic alias for now.
  BoxArt = "tw-aspect--3x4",
}

const AspectRatioValues = {
  [AspectRatio.Aspect21x9]: 21 / 9,
  [AspectRatio.Aspect16x9]: 16 / 9,
  [AspectRatio.Aspect4x3]: 4 / 3,
  [AspectRatio.Aspect1x1]: 1,
  [AspectRatio.Aspect3x4]: 3 / 4,
  [AspectRatio.Aspect3x2]: 3 / 2,
  [AspectRatio.BoxArt]: 3 / 4,
};

export enum Align {
  Bottom = 1,
  Center = 2,
  Top = 3,
}

interface ScAspectRatioProps {
  /**
   * Align the content within the component (if the content has a different
   * aspect ratio than the component).
   */
  align?: Align;
  /**
   * Show overflowing content (if the content has a different aspect ratio
   * than the component).
   */
  allowOverflow?: boolean;
}

function getAlign({ align }: ScAspectRatioProps): string {
  switch (align) {
    case Align.Bottom:
      return "bottom: 0;";
    case Align.Center:
      return "top: 50%; transform: translateY(-50%)";
    default:
      return "top: 0;";
  }
}

// @ts-ignore
export const ScAspectSpacer = styled.div<{ ratio: number | any }>`
  ${(props) => `padding-bottom: ${(1 / Math.abs(props.ratio)) * 100}%`}
`;

export const ScAspectRatio = styled.div<ScAspectRatioProps>`
  position: relative;
  width: 100%;
  overflow: ${(props) => (props.allowOverflow ? "visible" : "hidden")};

  /* stylelint-disable selector-max-universal, selector-max-type */
  & > *:not(${ScAspectSpacer}) {
    position: absolute;
    left: 0;
    width: 100%;
    min-height: 100%;
    ${getAlign}
  }
`;

export interface AspectProps extends ScAspectRatioProps {
  /**
   * The desired aspect ratio. If expressed as a number this value is width divided by height.
   * For example, a 3x4 ratio (box art) is calculated as 3 divided by 4 which equals 0.75
   *
   * @example 0.75
   */
  ratio?: AspectRatio | number;
  /**
   * The content to be contained within the aspect.
   *
   * @example <img src="https://static-cdn.jtvnw.net/s3_vods/twitchpresents/159083707/cf22b5ad-ce79-454b-bd91-c9831579ecfc/thumb/index-0000000000-1280x720.jpg" />
   */
  children?: ReactNode;
  /**
   * Deprecated alias for allowOverflow.
   *
   * @deprecated
   */
  overflow?: boolean;
}

/**
 * Aspect is used to set the height of an element based on its width for
 * applications such as box art and thumbnail images.
 */
export const Aspect: FC<AspectProps> = ({
  ratio = AspectRatioValues[AspectRatio.Aspect16x9],
  children,
  allowOverflow,
  overflow,
  ...aspectProps
}) => {
  return (
    <ScAspectRatio
      className="tw-aspect"
      allowOverflow={allowOverflow ?? overflow}
      {...aspectProps}
    >
      <ScAspectSpacer
        ratio={
          ratio && typeof ratio === "string" ? AspectRatioValues[ratio] : ratio
        }
      />
      {children}
    </ScAspectRatio>
  );
};

Aspect.defaultProps = {
  align: Align.Top,
  // This uses the deprecated prop to enable proper fallback from the new
  // `allowOverflow`. Once `overflow` is fully removed, this should switch to
  // `allowOverflow`.
  overflow: false,
};

Aspect.displayName = "Aspect";
