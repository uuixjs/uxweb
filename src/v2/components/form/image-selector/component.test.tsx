import {
  ImageSelector,
  ImageSelectorIconPosition,
  ImageSelectorProps,
  ImageSelectorType,
} from "./component";

import { AspectRatio } from "../../aspect";
import { BorderRadius } from "lib/ui-utils";
import { SVGAsset } from "../../svg";
import { setupShallowTest } from "../../../tests/helpers";

function getRequiredProps(): ImageSelectorProps {
  return {
    title: "Test Label",
    id: "test-id",
    src: "/testSRC",
    alt: "Test Alt",
    type: ImageSelectorType.Radio,
  };
}

function getOptionalProps(): ImageSelectorProps {
  return {
    autoFocus: true,
    checked: true,
    defaultChecked: true,
    disabled: true,
    error: true,
    id: "test-id",
    title: "Test Label",
    name: "Test name",
    onBlur: jest.fn(),
    onChange: jest.fn(),
    onFocus: jest.fn(),
    required: true,
    src: "/testSRC",
    alt: "Test Alt",
    tabIndex: 0,
    type: ImageSelectorType.Checkbox,
    value: "Test value",
    borderRadius: BorderRadius.Medium,
    hoverScale: true,
    selectedMask: true,
    selectedIcon: {
      position: ImageSelectorIconPosition.TopRight,
      icon: SVGAsset.Heart,
    },
    ratio: AspectRatio.Aspect16x9,
  };
}

const setupShallow = setupShallowTest(ImageSelector, getRequiredProps);

describe("ImageSelector", () => {
  it("renders checkbox component with required props", () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders checkbox component with all props", () => {
    const { wrapper } = setupShallow(getOptionalProps());
    expect(wrapper).toMatchSnapshot();
  });
});
