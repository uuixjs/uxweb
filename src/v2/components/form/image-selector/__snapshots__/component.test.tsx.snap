// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageSelector renders checkbox component with all props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`ImageSelector renders checkbox component with required props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
